frontend/src/vendor/profile/VendorProfileMarkets.tsx

```diff
@@ -18,24 +18,37 @@ const VendorProfileMarketsTitle = styled.h3`
 const EmptyPageMessage = styled.p`
   text-align: left;
   font-size: 12px;
+  margin: 0;
 `;
 
 const VendorProfileMarkets = () => {
-  const vendorId = useParams().id;
+  interface urlParam {
+    id: string;
+  }
+  const vendorId = useParams<urlParam>().id;
 
-  const { data: upcomingMarkets, isLoading } = useQuery(
+  const { data: upcomingMarkets, isLoading } = useQuery<any>(
     ["market", vendorId],
     () => fetchUpcomingMarketsByVendorId(vendorId)
   );
 
+  interface marketProps {
```

Comment:
Ezt szervezzük ki egy Market.ts-be és a neve lehet az interface-nek Market. A fentebb írt upcomingMarkets típusa pedig Market[] lehet, ilyenkor nem is kell már a lenti map-nél annotálni a függvény paramétert, mert ki fogja találni a TS.
A props postfixet egyébként akkor használjuk a típus nevében, ha a komponens props-ának a típusát jelöljük vele, egyébként nem szoktuk props-nak hívni, ha nem props.